core/src/main/java/com/yahoo/oak/MemoryManager.java

```diff
@@ -23,13 +23,16 @@
     long allocated();
 
     /**
-     * This method allocates memory out of the off-heap, i.e., the ByteBuffer inside of {@code s} is pointing to the
-     * off-heap. The blockID, is an internal reference to which block the ByteBuffer points, allowing the functions
-     * {@code getSliceFromBlockID} and {@code getByteBufferFromBlockID} to reconstruct the same ByteBuffer.
-     *  @param s        - an allocation object to update with the new allocation
+     * This method allocates memory out of the off-heap, i.e. off-heap cut of the given size in bytes.
```

Comment:
"cut" is a new phrase. What does it mean?

Reply:
yes, I have the feeling that Slice is an object that doesn't exactly explains the physical (virtual) memory addresses being references. Like, Slice can be duplicated, not associated, associated, deleted as an object, etc. while the memory addresses it is representing are still the same, and do not go through the same transformation.

Anyway, I feel that the memory itself and the Slice object need to be addresses with different words. "Off-heap cut" (and also Slice) are explained at the beginning of MemoryManager interface file.